src/aiq/builder/builder.py

```diff
@@ -108,6 +110,16 @@ async def get_llm(self, llm_name: str | LLMRef, wrapper_type: LLMFrameworkEnum |
     def get_llm_config(self, llm_name: str | LLMRef) -> LLMBaseConfig:
         pass
 
+    def get_object_store_clients(self, object_store_names: Sequence[str | ObjectStoreRef]) -> list[ObjectStore]:
+        """
+        Return a list of all object store clients.
+        """
+        return [self.get_object_store_client(name) for name in object_store_names]
+
+    @abstractmethod
+    def get_object_store_client(self, object_store_name: str | ObjectStoreRef) -> ObjectStore:
+        pass
```

Comment:
New APis on the builder object should all be async by default. Even if there isnt much of a reason, its better to start async than need to add it in the future.